docs/source/executing-operations/file-uploads.mdx

```diff
@@ -0,0 +1,278 @@
+---
+title: File uploads
+description: Receive file uploads with the Apollo Router
+minVersion: X.x  
+---
+
+<PreviewFeature />
+
+<EnterpriseFeature />
+
+Learn how to configure the Apollo Router to receive file uploads in client requests using the [GraphQL Multipart Request Spec](https://github.com/jaydenseric/graphql-multipart-request-spec).
+
+## About file uploads using multipart requests
+
+The [GraphQL Multipart Request Spec](https://github.com/jaydenseric/graphql-multipart-request-spec) enables you to send files using arguments in a GraphQL mutation.
+
+{/* To-do: specifics + examples */}
+
+## File upload configuration and usage
+
+To enable file uploads in clients, you must both [configure support in the Apollo Router](#configure-file-upload-support-in-the-router) and [ensure client usage conforms to requirements](#client-usage-requirements).
+
+### Configure file upload support in the router
+
+By default, receiving client file uploads is _not_ enabled in the Apollo Router.
+To enable file upload support, set the following fields in your `router.yaml` configuration file:
+
+```yaml title="router.yaml" showLineNumbers="false"
+preview_file_uploads:
+  enabled: true
+  protocols:
+    multipart:
+      enabled: true
+      mode: streaming
+      limits:
+        max_file_size: 5mb # default 1mb
+        max_files: 2 # default 4
+```
+
+#### Mode
+
+The only supported `mode` is `streaming`. This mode means that the router doesn't retain uploaded files in memory during a request.
+If a request cannot be fulfilled in a streaming fashion, the router returns the [`UPLOADS_OPERATION_CANNOT_STREAM`](#uploads_operation_cannot_stream) error.
```

Comment:
Let's add some brief guidance on what will definitely work for people. Something like:

To ensure that your operation is streamable, make sure to avoid nesting for file uploads.

For example (fixme):

```
Mutation {
  upload_file()
  upload_file()
  upload_file()
}
```

Reply:
Good call. Beyond nesting files, are we are of anything that would preclude an operation from being streamable? There's already a size limit and I'm not sure it's necessary to say anything abut network connections, so maybe not?

Reply:
Also, could you please check the now included example makes sense? TIA!

Reply:
It's more like the example you have is OK but further nesting is not recommended.
```
mutation($f1: File, $f2: File, $f3: File) {
    upload(data: $f1) {
        __typename
    }
    upload(data: $f2) {
        __typename
    }

    somethingComplex{
        upload(data: $f3) {
            __typename
        }
    }
}

```
`f1` and `f2` are OK. They are top level in the mutation.
`f3` is nested in something else, and is more risky.

Reply:
I see. Thank you!